superset-frontend/src/components/Menu/LanguagePicker.tsx

```diff
@@ -33,43 +34,72 @@ interface LanguagePickerProps {
   languages: Languages;
 }
 
+const StyledLabel = styled.div`
+  display: flex;
+  align-items: center;
+
+  & i {
+    margin-right: ${({ theme }) => theme.gridUnit}px;
+  }
+`;
+
+const StyledFlag = styled.div`
+  margin-top: 2px;
+`;
+
+const StyledIcon = styled(Icons.TriangleDown)`
+  ${({ theme }) => `
+    margin-top: -${theme.gridUnit}px;
+    margin-left: -${theme.gridUnit * 2}px;
+  `}
+`;
+
 export default function LanguagePicker({
   locale,
   languages,
 }: LanguagePickerProps) {
-  const [dropdownOpen, setDropdownOpen] = useState(false);
+  const theme = useTheme();
+  const [open, setOpen] = useState(false);
+
+  const options = Object.keys(languages).map(langKey => ({
+    label: (
+      <StyledLabel className="f16">
+        <i className={`flag ${languages[langKey].flag}`} />{' '}
+        {languages[langKey].name}
+      </StyledLabel>
+    ),
+    value: langKey,
+    flag: (
+      <StyledFlag className="f16">
+        <i className={`flag ${languages[langKey].flag}`} />
+      </StyledFlag>
+    ),
+  }));
 
   return (
-    <NavDropdown
-      onMouseEnter={() => setDropdownOpen(true)}
-      onMouseLeave={() => setDropdownOpen(false)}
-      onToggle={value => setDropdownOpen(value)}
-      open={dropdownOpen}
-      id="locale-dropdown"
-      title={
-        <span className="f16">
-          <i className={`flag ${languages[locale].flag}`} />
-        </span>
+    <Select
+      defaultValue={locale}
+      open={open}
+      onMouseEnter={() => setOpen(true)}
+      onMouseLeave={() => setOpen(false)}
+      onDropdownVisibleChange={open => setOpen(open)}
+      bordered={false}
+      options={options}
+      suffixIcon={
+        <StyledIcon
+          iconColor={theme.colors.grayscale.base}
+          className="ant-select-suffix"
+        />
       }
-      data-test="language-picker"
-    >
-      <Menu
-        onSelect={({ key }) => {
-          window.location.href = languages[key].url;
-        }}
-      >
-        {Object.keys(languages).map(langKey =>
-          langKey === locale ? null : (
-            <Menu.Item key={langKey}>
-              {' '}
-              <div className="f16">
-                <i className={`flag ${languages[langKey].flag}`} /> -{' '}
-                {languages[langKey].name}
-              </div>
-            </Menu.Item>
-          ),
-        )}
-      </Menu>
-    </NavDropdown>
+      listHeight={400}
+      dropdownAlign={{
+        offset: [-135, 0],
```

Comment:
could also do some styling to reposition this, e.g.:
```
transform: translateX(-100%);
```

Reply:
@rusackas I changed to make the component wrap big texts in multiple lines instead of using `transform` to avoid fat dropdowns 😉. Since the width is fixed, the offset is always correctly positioned. If you agree, you can remove the `need:followup` label.